### PR TITLE
Lineage improvements

### LINKED ISSUE #16649: Lineage Improvement
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**

Some UI issues on Lineage Pages

**To Reproduce**

Screenshots or steps to reproduce

- [x] Entity Column name is cut off.
image

- [x] Entity name is being cut while adding.
image

- [x]  On performing request description on the edge, the page is breaking

https://github.com/open-metadata/OpenMetadata/assets/66266464/aa1f4791-fcef-4526-9b3e-df57b2d855ba

- [x] Loading state on the save of edge modal

https://github.com/open-metadata/OpenMetadata/assets/66266464/7bb2ee00-bd0a-4e6d-b6f7-025e9926cf41

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### LINKED ISSUE #16653: Pipeline UI tooltips consume window for large lineage graphs
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**
In 1.4.2, when viewing the lineage tab for a pipeline, a tooltip-style pop up appears above the edge for the pipeline (even when not hovering over or clicking the edge). This behavior is seen for every pipeline edge between datasets, even when there are many of these edges in a graph on a single page. This means for a lineage graph with many edges representing one given pipeline, as the user scrolls around the graph, there are a large number of these pop ups clouding up the graph, leading to a degraded user experience.

**To Reproduce**
To reproduce, a user can navigate to a pipeline in OpenMetadata, and go to the lineage page. If the pipeline creates a large graph with many edges representing the given pipeline, the user will see as many pop ups all at once as there are relevant edges

**Expected behavior**
Given the pipeline already highlights and "blinks" the relevant pipeline icons on the edges, I don't really think the automatic tool-tip pop up is actually helpful to identify relevant edges until the user clicks or hovers over the edge. Otherwise if OM really wants to have the tooltip pop up unprompted, I would expect only one tooltip style popup at a time, or just on the initial load of the page, etc.

**Version:**
 - OpenMetadata version: 1.4.2

**Additional context**
Cannot share screenshots due to company PII but let me know if you have any follow up questions. The other lineage UI fixes between 1.4.0 and 1.4.2 have been great, thank you!

### LINKED ISSUE #16681: UI / Lineage - Mark deleted dashboard
**Is your feature request related to a problem? Please describe.**
When the **Mark Deleted Dashboards** is enabled, it's not possible to identify if the DS is deleted in the lineage view.

**Describe the solution you'd like**
Maybe, change the border color for deleted assets or add an icon.

Capture d’écran 2024-06-17 à 12 06 45

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Fixes #16681 
Fixes #16649 
Fixes #16653 

I worked on ... because ...

https://github.com/open-metadata/OpenMetadata/assets/33024356/fd29fbc0-7a24-445b-99b4-a4e39341f922

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+  await page.click('[data-testid="lineage-layer-btn"]');
+  await page.click('[data-testid="lineage-layer-column-btn"]');
+};
+
+export const performZoomOut = async (page: Page) => {
+  for (let i = 0; i < 5; i++) {
+    const zoomOutBtn = page.locator('.react-flow__controls-zoomout');
+    const enabled = await zoomOutBtn.isEnabled();
+    if (enabled) {
+      zoomOutBtn.dispatchEvent('click');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:43]:
```suggestion
      zoomOutBtn.click();
```

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+  page.click(`[data-testid="edge-${fromNodeFqn}-${toNodeFqn}"]`, {
+    force: true,
+  });
+
+  if (
+    ['Table', 'Topic'].indexOf(fromNode.getType()) > -1 &&
+    ['Table', 'Topic'].indexOf(toNode.getType()) > -1
+  ) {
+    await page.locator('[data-testid="add-pipeline"]').dispatchEvent('click');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:64]:
```suggestion
    await page.click('[data-testid="add-pipeline"]');
```

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    await page
+      .locator(
+        '[data-testid="add-edge-modal"] [data-testid="remove-edge-button"]'
+      )
+      .dispatchEvent('click');
+  } else {
+    await page.locator('[data-testid="delete-button"]').dispatchEvent('click');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:71]: same here, please replace everywhere!

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+  fromNode: EntityClass,
+  toNode: EntityClass
+) => {
+  const type = getEntityTypeSearchIndexMapping(toNode.type);
+  const fromNodeFqn = get(fromNode, 'entityResponseData.fullyQualifiedName');
+  const toNodeName = get(toNode, 'entityResponseData.name');
+  const toNodeFqn = get(toNode, 'entityResponseData.fullyQualifiedName');
+
+  await page.locator(`[data-testid="${type}-draggable-icon"]`).hover();
+  await page.mouse.down();
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:115]: Dont we need any locator here to perform this action?

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+  await page.locator('[data-testid="lineage-details"]').hover();
+  await page.mouse.up();
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:117]: same here

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+
+  await page.locator('[data-testid="suggestion-node"]').dispatchEvent('click');
+
+  const suggestionNodeInput = page.locator(
+    '[data-testid="suggestion-node"] input'
+  );
+  await suggestionNodeInput.dispatchEvent('click');
+  await suggestionNodeInput.fill(toNodeName);
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:125]: We can directly get element and type in it
```suggestion
 await page.locator(
    '[data-testid="suggestion-node"] input'
  ).fill(toNodeName);
```
or
```suggestion
 await page.fill(
    '[data-testid="suggestion-node"] input', toNodeName
  );
```

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    `lineage-node-${toNodeFqn}`
+  );
+};
+
+export const verifyNodePresent = (page: Page, node: EntityClass) => {
+  const nodeFqn = get(node, 'entityResponseData.fullyQualifiedName');
+  const name = get(node, 'entityResponseData.name');
+  const lineageNode = page.locator(`[data-testid="lineage-node-${nodeFqn}"]`);
+
+  expect(lineageNode).toBeVisible();
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:142]: await is missing

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+    .dispatchEvent('click');
+
+  await dragConnection(
+    page,
+    `lineage-node-${fromNodeFqn}`,
+    `lineage-node-${toNodeFqn}`
+  );
+};
+
+export const verifyNodePresent = (page: Page, node: EntityClass) => {
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:137]: this fn should be async

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+    page.getByTestId('asset-description-container').getByRole('paragraph')
+  ).toContainText(description);
+};
+
+const verifyPipelineDataInDrawer = async (
+  page: Page,
+  fromNode: EntityClass,
+  toNode: EntityClass,
+  pipelineItem: PipelineClass,
+  bVerifyPipelineLineage: boolean
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:224]: the purpose for variable `bVerifyPipelineLineage` is not very clear from the name

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+  const saveRes = page.waitForResponse('/api/v1/lineage');
+  await page.click('[data-testid="save-button"]');
+  await saveRes;
+};
+
+export const deleteNode = async (page: Page, node: EntityClass) => {
+  const nodeFqn = get(node, 'entityResponseData.fullyQualifiedName');
+  await page
+    .locator(`[data-testid="lineage-node-${nodeFqn}"]`)
+    .dispatchEvent('click');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:285]: I have seen this clicking method to lot of places, is there any use case for that?

**Review by ShaileshParmar11 — COMMENTED** [review]
added few comments

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 10 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:285]: Kept it explicitly. Refer this 
https://playwright.dev/docs/input#programmatic-click

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 1 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:43]: Kept it explicitly. Refer this
https://playwright.dev/docs/input#programmatic-click

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 2 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:64]: Kept it explicitly. Refer this
https://playwright.dev/docs/input#programmatic-click

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 3 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:71]: Kept it explicitly. Refer this
https://playwright.dev/docs/input#programmatic-click

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 4 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:115]: No we don't need it.
Refer this example from playwright - https://playwright.dev/docs/input#dragging-manually

File: openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts (hunk 5 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts:117]: No we don't need it.
Refer this example from playwright - https://playwright.dev/docs/input#dragging-manually

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**